File: update_paths.dart
import 'dart:io';

void main() {
  final file = File('lib/services/vocabulary_initializer.dart');
  var content = file.readAsStringSync();
  
  final regex = RegExp(r"VocabularyItem\([^)]+labels:\s*\{'en':\s*'([^']+)'\}[^)]+\)");
  
  final matches = regex.allMatches(content).toList();
  for (var match in matches) {
    final block = match.group(0)!;
    final label = match.group(1)!;
    if (!block.contains('imagePath:')) {
      final replacement = block.replaceFirst(
        'id: _uuid.v4(),', 
        "id: _uuid.v4(),\n        imagePath: 'assets/images/${label.replaceAll(' ', '_')}.jpg',"
      );
      content = content.replaceAll(block, replacement);
    }
  }
  
  file.writeAsStringSync(content);
}

File: update_paths.py
import re

file_path = r'c:\Users\fortn\Desktop\awaz\lib\services\vocabulary_initializer.dart'
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

def replacer(match):
    labels_match = re.search(r"labels:\s*\{'en':\s*'([^']+)'\}", match.group(0))
    if labels_match:
        label = labels_match.group(1)
        if 'imagePath:' not in match.group(0):
            insert_str = f"\n        imagePath: 'assets/images/{label.replace(' ', '_')}.jpg',"
            return match.group(0).replace('id: _uuid.v4(),', 'id: _uuid.v4(),' + insert_str)
    return match.group(0)

new_content = re.sub(r'VocabularyItem\([^)]+\)', replacer, content)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(new_content)
